algorithms.md

```diff
@@ -26,7 +26,7 @@ code here
 ```python
 # general language-independent implementation
 def find_substr(string, substr):
-    for i in range(0, len(string)):
+    for i in range(0, len(string) - len(substring) + 1):
```

Comment:
Of course we can add an extra step like `if substr == '': return True`, but the current code also checks this. Without an iteration we can not reach `return True`.
This `+1` is checked by a case with two non-empty strings with same length: we really need one iteration to compare them.

BTW, there is a typo in the patch: I used `substring` but `substr` is right. I will update it soon.